routes/web.php

```diff
@@ -29,6 +29,10 @@
     // for Profile
     Route::get('/profile/{id}/show', [ProfileController::class, 'showProfile'])->name('profile.show');
     Route::get('/profile/edit', [ProfileController::class, 'editProfile'])->name('profile.edit');
+    Route::patch('profile/update', [ProfileController::class, 'updateProfile'])->name('profile.update');
+    Route::post('profile/password', [ProfileController::class, 'changePassword'])->name('profile.password');
+    Route::delete('/profile/{id}/deactivate', [ProfileController::class, 'deactivate'])->name('profile.deactivate');
```

Comment:
No need to add {$id} because you're not using it in your deactivate function
You can remove the , User $user since you query it using the Auth::user()->id
 ```
public function deactivate(Request $request, User $user)
    {
        $user = $this->user->findOrFail(Auth::user()->id);
        $user->delete();
        return redirect()->route('homepage');
    }
```

Reply:
Try this also
```
public function deactivate(Request $request)
{
    $user = Auth::user();
    $user->delete();
    return redirect()->route('homepage');
}
```

Reply:
@Ninit29 
Thank you for your suggestion. I remembered the role. I removed {id}, so kindly confirm it again please.

Reply:
I modified it to `public function deactivate(Request $request)
{
    $user = Auth::user();
    $user->delete();
    return redirect()->route('homepage');
}`

Reply:
I added notification messages.